src/resolver.ts

```diff
@@ -59,12 +59,30 @@ export interface ParsedDID {
   params?: Params
 }
 
-export type DIDResolver = (did: string, parsed: ParsedDID, didResolver: Resolver) => Promise<null | DIDDocument>
+export type DIDResolver = (did: string, parsed: ParsedDID, resolver: Resolver) => Promise<null | DIDDocument>
+export type WrappedResolver = () => Promise<null | DIDDocument>
+export type DIDCache = (parsed: ParsedDID, resolve: WrappedResolver) => Promise<null | DIDDocument>
 
 interface ResolverRegistry {
   [index: string]: DIDResolver
 }
 
+export function inMemoryCache() : DIDCache {
+  const cache: Map<String, DIDDocument|null> = new Map()
```

Comment:
This should use the primitive type `string` instead of `String`
```suggestion
  const cache: Map<string, DIDDocument|null> = new Map()
```